kafkabp/config.go

```diff
@@ -0,0 +1,23 @@
+package kafkabp
+
+import "github.com/Shopify/sarama"
+
+// ConsumerConfig is the configuration struct passed to NewConsumer.
+type ConsumerConfig struct {
+	// Required.
+	Brokers []string
+	Topic   string
+
+	// Optional. Defaults to DefaultSaramaConfig.
+	SaramaConfig *sarama.Config
+
+	// Optional. If set, overrides the client ID set in SaramaConfig.
+	ClientID string
+
+	// Optional. Defaults to OffsetOldest.
+	Offset KafkaOffset
```

Comment:
For getting this from YAML, we might want to make it a `string` and have it support 2 values, "oldest" and "newest" and then convert that to the appropriate sarama value?

Reply:
Ooh, that's a better idea than the typedef!

Reply:
note that you don't have to use string type to do that. as long as the typedef implemented [`yaml.Unmarshaler`](https://pkg.go.dev/gopkg.in/yaml.v2#Unmarshaler) it would work.

but yaml v2's `Unmarshaler` interface is defined so confusing I have no idea how it's supposed to work even after reading their code. [v3's `Unmarshaler`](https://pkg.go.dev/gopkg.in/yaml.v3#Unmarshaler) is much better.